### PR TITLE
move Labels from array to map to be consistent with docker

### PR DESCRIPTION
**Pull request by mavenugo**
While working on the updating the network API in docker, i realized that the labels are consistently
used as map[string]string in containers and images. Its the UX that converts the []string to map[string]string.
Network being a docker top level construct must also adhere to that consistency.

Signed-off-by: Madhu Venugopal <madhu@docker.com>

### CODE REVIEW

File: api/types.go [hunk 1]
```diff
@@ -34,9 +34,9 @@ type sandboxResource struct {
 
 // networkCreate is the expected body of the "create network" http request message
 type networkCreate struct {
-	Name        string   `json:"name"`
-	NetworkType string   `json:"network_type"`
-	Labels      []string `json:"labels"`
+	Name        string            `json:"name"`
+	NetworkType string            `json:"network_type"`
+	Labels      map[string]string `json:"labels"`
```
Comment (aboch) [inline api/types.go]: `Labels` is optional. I noticed we do not nil check before calling `libnetwork.NetworkOptionLabels(create.Labels)` .
(Check should be done in api.go, not in the option setter itself.)
Should we also add the `json:",omitempty"` tag ?

Comment (mavenugo) [inline api/types.go]: No. folks prefer to see empty labels :) thats the comment I got in docker side.

Comment (mavenugo) [inline api/types.go]: @aboch done. PTAL

**Comment by aboch** [comment]
LGTM

**Comment by mrjana** [comment]
LGTM

**Pull request merged**